rali/rali/source/rali_api_meta_data.cpp

```diff
@@ -191,6 +218,43 @@ RALI_API_CALL raliGetBoundingBoxLabel(RaliContext p_context, int* buf)
     }
 }
 
+void
+RALI_API_CALL raliGetOneHotImageLabels(RaliContext p_context, int* buf, int numOfClasses)
+{
+    auto context = static_cast<Context*>(p_context);
+    auto meta_data = context->master_graph->meta_data();
+    if(!meta_data.second) {
+        WRN("No label has been loaded for this output image")
+        return;
+    }
+    size_t meta_data_batch_size = meta_data.second->get_label_batch().size();
+    if(context->user_batch_size() != meta_data_batch_size)
+        THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
+
+    int labels_buf[meta_data_batch_size];
+    int one_hot_encoded[meta_data_batch_size*numOfClasses];
+    memset(one_hot_encoded, 0, sizeof(int) * meta_data_batch_size * numOfClasses);
+    memcpy(labels_buf, meta_data.second->get_label_batch().data(),  sizeof(int)*meta_data_batch_size);
+    int label_index;
```

Comment:
move label_index to inside for loop where it is used

Reply:
Moved label_index inside loop